.github/workflows/ci.yml

```diff
@@ -101,35 +104,36 @@ jobs:
       - name: Run linters
         run: poetry run invoke lint
 
-
-  deploy:
-    name: Deploy
+  prep-release:
+    name: Release preparations
     environment: Deployment
     needs: [test, lint]
     runs-on: ubuntu-latest
     if: ${{ github.ref=='refs/heads/main' && github.event_name!='pull_request' }}
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Setup Python
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v4
         with:
           python-version: "3.8"
       - name: Check release
         id: check_release
         run: |
           python -m pip install --upgrade pip
-          python -m pip install poetry githubrelease httpx==0.16.1 autopub
-          echo "##[set-output name=release;]$(autopub check)"
-      - name: Publish
+          python -m pip install poetry autopub[github]
+          echo "release=$(autopub check)" >> "$GITHUB_OUTPUT"
+          EOF=$(dd if=/dev/urandom bs=15 count=1 status=none | base64)
+          echo "release<<$EOF" >> "$GITHUB_OUTPUT"
```

Comment:
`set-ouput` is deprecated, I have switched it over to multiline strings since we will never know how much `autopub check` will output… If `autopub check` where to write into `$GITHUB_OUTPUT/STATE` on it's own, we wouldn't need to handle this extra. I mean it has special handling for circle ci also… https://github.com/autopub/autopub/blob/8d41aed1ca9ff0aa197a7850eb7d4314a96a1206/autopub/check_release.py#L13

Reply:
The intention was always for AutoPub to internally handle as much of the machinery as possible, so I agree that AutoPub should be enhanced to obviate the need to handle this in GitHub Actions. 👍